src/targets/maven.ts

```diff
@@ -73,6 +78,10 @@ export class MavenTarget extends BaseTarget {
     super(config, artifactProvider);
     this.mavenConfig = this.getMavenConfig();
     this.checkRequiredSoftware();
+
+    if (process.env.GPG_PRIVATE_KEY) {
+      importGPGKey(process.env.GPG_PRIVATE_KEY);
+    }
```

Comment:
This check is already done in `getTargetSecrets`, line 102. At this point, if the env var exists it's available in the target config, so you should be able to just import it.

On the other hand, what's the error message if the `importGPGKey` fails? Is it descriptive enough?

Reply:
`getTargetSecrets` is making them required, where this is optional.